actors/core/src/main/java/cloud/orbit/actors/annotation/Timeout.java

```diff
@@ -0,0 +1,49 @@
+/*
+ Copyright (C) 2017 Electronic Arts Inc.  All rights reserved.
+
+ Redistribution and use in source and binary forms, with or without
+ modification, are permitted provided that the following conditions
+ are met:
+
+ 1.  Redistributions of source code must retain the above copyright
+     notice, this list of conditions and the following disclaimer.
+ 2.  Redistributions in binary form must reproduce the above copyright
+     notice, this list of conditions and the following disclaimer in the
+     documentation and/or other materials provided with the distribution.
+ 3.  Neither the name of Electronic Arts, Inc. ("EA") nor the names of
+     its contributors may be used to endorse or promote products derived
+     from this software without specific prior written permission.
+
+ THIS SOFTWARE IS PROVIDED BY ELECTRONIC ARTS AND ITS CONTRIBUTORS "AS IS" AND ANY
+ EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ DISCLAIMED. IN NO EVENT SHALL ELECTRONIC ARTS OR ITS CONTRIBUTORS BE LIABLE FOR ANY
+ DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package cloud.orbit.actors.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Defines a custom timeout for the given actor method that will be used instead of the
+ * configured <code>responseTimeoutMillis</code>.
+ *
+ * @author Johno Crawford (johno@sulake.com)
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface Timeout
+{
+    int ttlDuration();
+    TimeUnit ttlUnit() default TimeUnit.SECONDS;
```

Comment:
Should this be timeout instead of ttl?

Reply:
If I can give my 2 cents... from my perspective, a "timeout" is used when you have a reaction, "throws an exception". "TTL" I think naturally means something just expired and it will be discarded.

Reply:
I think we should just go with `value` and `timeUnit` like the new `TimeToLive` annotation, that way we avoid the naming question entirely.

Reply:
By the way, I dont know if this is already being done, but I think it would be a good idea to store the discarded count (by class/method?) in a metric, So we can see exactly what is happening when the system is under pressure. Just because there are messages that can be discarded, this does not mean we dont need to know when this is happening. :)

Reply:
This will just fail like it does today, so your Task will contain a timeout exception instead of a result. What you do with that is up to you.

Reply:
Renamed to the same as TimeToLive annotation.